File: Zeynep Dilmaç/3.hafta/quicksort.py
#yapamadım..
def quicksort(arr,s,l):
    if s<l:
        p=partition(arr,s,l)
        quicksort(arr,s,p-1)
        quicksort(arr,p+1,l)
def partition(arr,p,l):
    pivot=arr[p]
    x=l+1
    for i in range(p+1,l+1):
        if(arr[i]>pivot):
            x -= 1
            (arr[i],arr[x])=(arr[x],arr[i])
    (arr[p],arr[x])=(arr[x],arr[p])
    return x
list1=[2,5,1,4,8,3]
quicksort(list1,0,len(list1)-1)
print(list1)

File: Yiğit Emir İşıkçı/week3/QuickSort.java
public class QuickSort {

    public static void quickSort(int[] arr, int lo, int hi){
        if(lo<hi){
            int q = partition(arr,lo,hi);
            quickSort(arr,lo,q-1);
            quickSort(arr,q+1,hi);
        }
    }
    public static int partition(int[] arr, int lo, int hi){
        int pivot = arr[lo];
        int j = hi;
        int i = lo+1;
        while(i<j){
            if(arr[i]>pivot && arr[j]<pivot){
                int temp1 = arr[i];
                arr[i] = arr[j];
                arr[j] = temp1;
                i++;j--;
                continue;
            }
            if(arr[i]<pivot){
                i++;
            }
            if(arr[j]>pivot){
                j--;
            }

        }
        if(pivot<arr[i] && pivot<arr[j]){
            int temp2 = arr[j-1];
            arr[j-1] = arr[lo];
            arr[lo] = temp2;
            return j-1;
        }

        int temp2 = arr[j];
        arr[j] = arr[lo];
        arr[lo] = temp2;
        return j;

    }

    public static void main(String[] args) {
        int[] arr = {8,6,5,4,10,1,3,2,-1,-123,12,11};
        quickSort(arr,0,11);
        for(int i:arr)  {
            System.out.println(i);
        }
    }
}
